transcendence-app/src/shared/guards/ws-two-factor-authenticated.guard.ts

```diff
@@ -0,0 +1,37 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { WsAuthenticatedGuard } from './ws-authenticated.guard';
+import { User } from '../../user/infrastructure/db/user.entity';
+import { ConfigService } from '@nestjs/config';
+import { EnvironmentVariables } from '../../config/env.validation';
+import { AuthorizationService } from '../../authorization/authorization.service';
+import { CaslAbilityFactory } from '../../authorization/casl-ability.factory';
+import {
+  IS_TWO_FACTOR_AUTHENTICATED_COOKIE_NAME,
+  IS_TWO_FACTOR_AUTHENTICATED_COOKIE_VALUE,
+} from '../constants';
+
+@Injectable()
+export class WsTwoFactorAuthenticatedGuard extends WsAuthenticatedGuard {
+  constructor(
+    private readonly configService: ConfigService<EnvironmentVariables>,
+    authorizationService: AuthorizationService,
+    caslAbilityFactory: CaslAbilityFactory,
+  ) {
+    super(authorizationService, caslAbilityFactory);
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const client = context.switchToWs().getClient();
+    const request = client.request;
+    const user: User | null = request.user;
+    const isAuth = await super.canActivate(context);
+
+    if (!user?.isTwoFactorAuthenticationEnabled) {
+      return isAuth;
+    }
+    const isTwoFactorAuthenticated =
+      request.signedCookies[IS_TWO_FACTOR_AUTHENTICATED_COOKIE_NAME] ===
+      IS_TWO_FACTOR_AUTHENTICATED_COOKIE_VALUE;
+    return isAuth && isTwoFactorAuthenticated;
+  }
```

Comment:
Same could be extended for the other guards we have, they are also splitted into http and ws

Reply:
I know we're repeating code, but I also know gbudau tried in the past, and failed. Now, we can keep trying, but it's outside of the scope of the ticket. Also gbudau is having a hard time lately with his workstation setup. I would open a ticket, and prioritize it, but not block this PR, because of this.

Reply:
Done, thank you both!

Reply:
Thanks @gbudau !